ARCs/arc-0014.md

```diff
@@ -0,0 +1,246 @@
+- **Algorand address**: the public key *PKa* of an Algorand account;
+- **User**: an Algorand account holder;
+- **Verifier**: a *system* that need to verify the identity of a User;
+- **dApp**: a decentralized Algorand application that natively runs on the Algorand blockchain, aka "*smart contract*",
+- **Wallet**: an off-chain application that stores the secret keys *SKa*s of Algorand accounts and can display and sign transactions for these accounts;
+- **message**: a generic string of bytes;
+- **digital signature**: a message signed with the private key of a blockchain identity, and specifically in this ARC with the *SKa*;
+
+## Motivation
+
+Traditional systems enforce SSO login leveraging on authentication of users with *credentials*, i.e. username and password. Once the identity of a user is verified, the system establishes an authenticated session. Sessions allows users to interact with the system without authenticate themselves anytime.
+
+> ARC-14 does not define any standard for session handling
+
+In a blockchain context, users do not have credentials. Conversely, in the Web3 users are identified through their unique blockchain public addresses (accounts for Algorand). Therefore, traditional authentication mechanisms result impractical. To interact with a dApp, users connect their wallet to the application. Using an authentication layer based on credentials on top of that mechanism might result redundant and inefficient.
+
+In Web3, dApps and traditional systems will be increasingly more interconnected. It is not difficult to imagine users consuming services both from a dApp and a traditional system simultaneously. In that case, a user should be authenticated first through traditional SSOs, and then connecting their blockchain wallet. A better approach should enforce a single SSO procedure.
+
+This ARC provides the standard to authenticate users leveraging on their Algorand accounts, and without relying on credentials verification.
+
+## Specification
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href="https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>
+> Comments like this are non-normative.
+
+Interfaces are defined in TypeScript. All the objects that are defined are valid JSON objects, and all JSON string types are UTF-8 encoded.
+
+This ARC uses interchangeably the terms "*blockchain address*", "*public key*", and "*PK*" to indicate the on-chain address of a blockchain identity, and in particular of an Algorand account.
+
+### Overview
+
+This document describes a standard approach to authenticate Users with their blockchain identity. In particular, in this ARC, Algorand address are used as a *unique identifiers*, and the secret key of that address is used to digitally sign a *message* as a proof of identity for a  Verifier.
```

Comment:
Grammatically better:
`... an Algorand address is used as a *unique identifier*`